client/src/raw.rs

```diff
@@ -0,0 +1,419 @@
+/// metric labels or whatever else to requests
+#[async_trait::async_trait]
+pub trait WorkflowService<T>: RawClientLike<T>
+where
+    T: tonic::client::GrpcService<tonic::body::BoxBody> + Send + 'static,
+    T::ResponseBody: tonic::codegen::Body + Send + 'static,
+    T::Error: Into<tonic::codegen::StdError>,
+    T::Future: Send,
+    <T::ResponseBody as tonic::codegen::Body>::Error: Into<tonic::codegen::StdError> + Send,
+{
+    proxy!(
+        register_namespace,
+        RegisterNamespaceRequest,
+        RegisterNamespaceResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
+            r.extensions_mut().insert(labels);
+        }
+    );
+    proxy!(
+        describe_namespace,
+        DescribeNamespaceRequest,
+        DescribeNamespaceResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
+            r.extensions_mut().insert(labels);
+        }
+    );
+    proxy!(
+        list_namespaces,
+        ListNamespacesRequest,
+        ListNamespacesResponse
+    );
+    proxy!(
+        update_namespace,
+        UpdateNamespaceRequest,
+        UpdateNamespaceResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
+            r.extensions_mut().insert(labels);
+        }
+    );
+    proxy!(
+        deprecate_namespace,
+        DeprecateNamespaceRequest,
+        DeprecateNamespaceResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
+            r.extensions_mut().insert(labels);
+        }
+    );
+    proxy!(
+        start_workflow_execution,
+        StartWorkflowExecutionRequest,
+        StartWorkflowExecutionResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
+            r.extensions_mut().insert(labels);
+        }
+    );
+    proxy!(
+        get_workflow_execution_history,
+        GetWorkflowExecutionHistoryRequest,
+        GetWorkflowExecutionHistoryResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
+            r.extensions_mut().insert(labels);
+        }
+    );
+    proxy!(
+        poll_workflow_task_queue,
+        PollWorkflowTaskQueueRequest,
+        PollWorkflowTaskQueueResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
+            r.extensions_mut().insert(labels);
+            r.set_timeout(LONG_POLL_TIMEOUT);
+        }
+    );
+    proxy!(
+        respond_workflow_task_completed,
+        RespondWorkflowTaskCompletedRequest,
+        RespondWorkflowTaskCompletedResponse,
+        |r| {
+            let labels = AttachMetricLabels::new(vec![namespace_kv(r.get_ref().namespace.clone())]);
```

Comment:
Should the task queue be attached here (and other calls)?

Reply:
Ya, may as well